internal/certigo/lib/cert_key_format.go

```diff
@@ -0,0 +1,45 @@
+package lib
+
+import (
+	"strings"
+)
+
+func NewCertificateKeyFormat(fileFormat string) CertificateKeyFormat {
+	fileFormat = strings.ToUpper(fileFormat)
+	switch fileFormat {
+	case "":
+		return CertKeyFormatNONE
+	case "PEM":
+		return CertKeyFormatPEM
+	case "DER":
+		return CertKeyFormatDER
+	case "JCEKS":
+		return CertKeyFormatJCEKS
+	case "PKCS12", "P12":
+		return CertKeyFormatPKCS12
+	default:
+		return CertKeyFormatNONE
+	}
+}
+
+type CertificateKeyFormat string
+
+const (
+	CertKeyFormatNONE CertificateKeyFormat = ""
+	// The file contains plain-text PEM data
+	CertKeyFormatPEM CertificateKeyFormat = "PEM"
+	// The file contains X.509 DER encoded data
+	CertKeyFormatDER CertificateKeyFormat = "DER"
+	// The file contains JCEKS keystores
+	CertKeyFormatJCEKS CertificateKeyFormat = "JCEKS"
+	// The file contains PFX data describing PKCS#12
+	CertKeyFormatPKCS12 CertificateKeyFormat = "PKCS12"
+)
+
+func (f *CertificateKeyFormat) Set(fileFormat string) {
+	*f = NewCertificateKeyFormat(fileFormat)
+}
+
+func (f CertificateKeyFormat) IsNone() bool {
+	return f == CertKeyFormatNONE
+}
```

Comment:
This seems misleading.. these are called `CertificateKeyFormat`, but as far as I can tell the only supported key format is PEM, whereas the cert formats are what vary.

Reply:
Both JCEKS and PKCS12 formats include both key and certificate files. 

The key file can also be in DER format, but its usage frequency is low, so it is currently not supported. 

I am not using the JCEKS format at the moment and haven't tested it, so the code for it is included but not made available for public use.